Objects/typeobject.c

```diff
@@ -8867,12 +8867,12 @@ super_init_without_args(PyFrameObject *f, PyCodeObject *co,
     }
 
     PyTypeObject *type = NULL;
-    for (i = 0; i < co->co_nfreevars; i++) {
-        PyObject *name = PyTuple_GET_ITEM(co->co_freevars, i);
+    i = co->co_nlocals + co->co_ncellvars;
+    for (; i < co->co_nlocalsplus; i++) {
+        PyObject *name = PyTuple_GET_ITEM(co->co_fastlocalnames, i);
         assert(PyUnicode_Check(name));
         if (_PyUnicode_EqualToASCIIId(name, &PyId___class__)) {
```

Comment:
Put an extra check here to test whether the variable is a free variable; it is much faster than calling `_PyUnicode_EqualToASCIIId`

I realize that this will be replaced when we add `MAKE_CELL`, but let's minimize the performance impact for now.

Reply:
We're already looping through only the free vars, so all of the entries will match `co->co_fastlocalkinds[i] & CO_FAST_FREE`.  I'll add an assert to make that clear.